docs/sources/flow/reference/components/otelcol.processor.attributes.md

```diff
@@ -0,0 +1,582 @@
+
+    * `key` is required. It specifies the attribute to act upon.
+    * One of `value`, `from_attribute` or `from_context` is required.
+    * `action = "insert"` is required.
+
+* `update`: Updates an attribute in input data where the key does exist.
+
+    * `key` is required. It specifies the attribute to act upon.
+    * One of `value`, `from_attribute` or `from_context` is required.
+    * `action = "update"` is required.
+
+* `upsert`: Performs insert or update. Inserts a new attribute in input data where the key does not already exist and updates an attribute in input data where the key does exist.
+
+    * `key` is required. It specifies the attribute to act upon.
+    * One of `value`, `from_attribute` or `from_context` is required:
+        * `value` specifies the value to populate for the key.
+        * `from_attribute` specifies the attribute from the input data to use to populate
+        the value. If the attribute doesn't exist, no action is performed.
+        * `from_context` specifies the context value to use to populate the attribute value.
+        If the key is prefixed with `metadata.`, the values are searched
+        in the receiver's transport protocol additional information like gRPC Metadata or HTTP Headers. 
+        If the key is prefixed with `auth.`, the values are searched
+        in the authentication information set by the server authenticator. 
+        Refer to the server authenticator's documentation part of your pipeline 
+        for more information about which attributes are available.
+        If the key doesn't exist, no action is performed.
+        If the key has multiple values the values will be joined with `;` separator.
+    * `action = "upsert"` is required.
+
+* `hash`: Hashes (SHA1) an existing attribute value.
+
+    * `key` and/or `pattern` is required.
+    * `action = "hash"` is required.
+
+* `extract`: Extracts values using a regular expression rule from the input key to target keys specified in the rule. If a target key already exists, it will be overridden. Note: It behaves similar to the Span Processor `to_attributes` setting with the existing attribute as the source.
+
+    * `key` is required. It specifies the attribute to extract values from. The value of `key` is NOT altered.
+    * `pattern` is required. It is the regex pattern used to extract attributes from the value of `key`. The submatchers must be named. If attributes already exist, they will be overwritten.
+    * `action = "extract"` is required.
+
+* `convert`: Converts an existing attribute to a specified type.
+
+    * `key` is required. It specifies the attribute to act upon.
+    * `action = "convert"` is required.
+    * `converted_type` is required and must be one of int, double or string.
+
+* `delete`: Deletes an attribute from the input data.
+
+    * `key` and/or `pattern` is required. It specifies the attribute to act upon.
+    * `action = "delete"` is required.
+
+### include/exclude blocks
```

Comment:
Also, title "include/exclude blocks" could be misinterpreted as "how to include or exclude blocks".

Reply:
> I don't want to have a ### include block and a ### exclude block separately, and copy paste the same contents.

I would prefer having both an `include` and `exclude` block; we should be optimizing for the reading experience over the writing experience. 

Instead of copying and pasting, you can put the content in docs/shared and use the docs/shared shortcode to deduplicate.

Reply:
It is not trivial to reuse code by putting it in `docs/shared`, because the "include block" and "exclude block" sections contain references to blocks such as `[attribute][]` and `[resource][]`. 

If we were to put the shared code in `docs/shared`, we could do one of those things:
1. Unlink the links in the shared code and leave them as text. This means changing `[attribute][]` to `attribute`. I think this wouldn't be a good solution, because the `attribute` block is described in the "Blocks" section. So the config in `docs/shared` for the "include block" would not be self sufficient.
2. Move the text with links to the "Blocks" section. This also wouldn't make sense, since the attributes in the "include" block are not mentioned yet. 
3. Move only the table to a shared md file, to avoid some repetition. But copy paste the bullet points about "One of the following is also required:"

Basically the problem is that in the "include block" description, we have to mention both its attributes and its blocks. But because the blocks definition is not in the "include block, it is hard to reuse md code.

For now I'll go with option 3, but please let me know if there is a better way.

The fact that blocks and attributes are defined separately in config suggests that they should be described separately, but in this case I cannot think of a reason why describing them in one place is wrong, and I can't think of a way to describe them separately.